src/main/java/org/springframework/data/elasticsearch/core/convert/MappingElasticsearchConverter.java

```diff
@@ -386,6 +389,11 @@ public void afterPropertiesSet() {
 
 		Map<String, Object> target = new LinkedHashMap<>();
 		for (Entry<String, Object> entry : source.entrySet()) {
+			
+			if(null == entry.getValue()) {
+				target.put(entry.getKey(),null);
+				continue ;
+			}
 
 			if (isSimpleType(entry.getValue())) {
```

Comment:
```suggestion
			
			if(entry.getValue() == null) {
				target.put(entry.getKey(),null);
			} else if (isSimpleType(entry.getValue())) {
```
rather than using continue, go on with an else if